dev/breeze/src/airflow_breeze/cache.py

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+from typing import Any, List, Tuple
+
+from airflow_breeze import global_constants
+from airflow_breeze.breeze import get_airflow_sources_root
+
+# This file is to read the cached data in .build folder
+BUILD_CACHE_DIR = os.path.join(get_airflow_sources_root, '.build')
+
+
+def read_from_cache_file(param_name: str):
+    parameter_value = ""
+    parameter_value_file = os.path.join(BUILD_CACHE_DIR, f".{param_name}")
+    with open(parameter_value_file) as value:
+        parameter_value = value.read()
+    return parameter_value
+
+
+def write_to_cache_file(param_name: str, param_value: str, check_allowed_parameter: bool = True):
+    allowed = False
+    if check_allowed_parameter:
+        allowed, allowed_values = check_if_parameter_allowed(param_name, param_value)
+    if (check_allowed_parameter is False) or ((check_allowed_parameter is True) and allowed):
+        # Then write to cache file after (verifying and allowed) or (without any verification)
```

Comment:
No need to comment./ especially if the if statement is simplfied. We should aim for self-explanatory code if possible (not always). The number of comments in the original bash were much bigger as the code there was not really obvious due to bash limitations, but we can do better with Python